rails_application/app/services/result.rb

```diff
@@ -0,0 +1,14 @@
+class Result
+  attr_reader :status, :args
+
+  def initialize(status, *args)
+    @status = status
+    @args = args
+  end
+
+  def path(name, &block)
+    return unless @status == name.to_sym
+
+    block.call(*@args)
+  end
```

Comment:
There is also an alternative implementation which uses more metaprogramming but results in a nicer DSL:

```
def method_missing(name, *_, &block)
  return unless @status == name.to_sym

  block.call(*@args)
end
```
and then the DSL looks like this:
```
MyService.call(params).then do |on|
  on.success { puts "Awesome!" }
  on.validation_failed { |errors| puts "Oh no, there are some errors: #{errors}" }
  on.something_else { |arg1, arg2| puts arg1 * arg2 }
end
```
Let me know which one you prefer :)